### PR TITLE
VMware: Refactor vmware_cluster

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* Update cluster logic
* Refactor code
* Documentation update

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Feature Pull Request
 - Docs Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_cluster.py
lib/ansible/module_utils/vmware.py

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_cluster.py [hunk 1]
```diff
@@ -148,113 +334,210 @@ def state_create_cluster(self):
...
+                das_config.hostMonitoring != self.params.get('ha_host_monitoring') or \
+                das_config.admissionControlPolicy.failoverLevel != self.params.get('ha_failover_level') or \
+                das_config.admissionControlEnabled != self.params.get('ha_admission_control_enabled') or \
+                das_config.defaultVmSettings.restartPriority != self.params.get('ha_restart_priority') or \
+                das_config.defaultVmSettings.vmToolsMonitoringSettings.vmMonitoring != self.params.get('ha_vm_monitoring') or \
+                das_config.defaultVmSettings.vmToolsMonitoringSettings.failureInterval != self.params.get('ha_vm_failure_interval') or \
+                das_config.defaultVmSettings.vmToolsMonitoringSettings.minUpTime != self.params.get('ha_vm_min_up_time') or \
+                das_config.defaultVmSettings.vmToolsMonitoringSettings.maxFailures != self.params.get('ha_vm_max_failures') or \
+                das_config.defaultVmSettings.vmToolsMonitoringSettings.maxFailureWindow != self.params.get('ha_vm_max_failure_window'):
+            ret = True
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_cluster.py]: Wondering what the advantage is to do it this way, or just have:
```python
if ...:
    return True
return False
```

**Review by pdellaert — APPROVED** [review]
Just a general wonder. And personally, i'd like to have the DRS and HA specific settings be sub options, but i'm not going to make that a blocker :)

**Comment by Akasurde** [comment]
rebuild_merge

**Pull request merged**